docs/src/tutorials/pluscal-tutorial.md

```diff
@@ -0,0 +1,241 @@
+This specification has been checked with the model checker [TLC][].
+Moreover, [Leslie Lamport][] has proved safety of this algorithm with
+the [TLAPS][].
+
+
+## Step 0: Remove the TLAPS proof
+
+Since we are not interested in the TLAPS proof, we copy [Bakery.tla][] to
+[BakeryWoTlaps.tla][] and modify it as follows:
+
+ - Remove `TLAPS` from the list of extended modules
+
+    ```tla
+    EXTENDS Naturals
+    ```
+
+ - Remove the theorem and its proof:
+
+    ```tla
+    THEOREM Spec => []MutualExclusion
+    ...
+    ```
+
+Alternatively, you can download the module `TLAPS.tla`.
+
+## Step 1: Add a module with type annotations
+
+Let us check the types of [BakeryWoTlaps.tla][] with Apalache:
+
+```sh
+$ apalache-mc typecheck BakeryWoTlaps.tla
+...
+Typing input error: Expected a type annotation for VARIABLE max
+```
+
+The type checker complains about missing type annotations. See the [Tutorial on
+Snowcat][] for details. When we try to add type annotations to the variables,
+we run into an issue. Indeed, the variables are declared with the Pluscal
+syntax:
+
+```tla
+{{#include ../../../test/tla/bakery-pluscal/BakeryWoTlaps.tla:71:74}}
+```
+
+The most straightforward approach would be to add type annotations directly in
+the Pluscal code. As reported in [Issue 1412][], this does not work as
+expected, as the Pluscal translator erases the comments.
+
+A simple solution is to add type annotations directly to the declarations in
+the generated TLA+ code. However, this solution is volatile. If we change the
+Pluscal code, our annotations will get overridden. We propose another solution
+that is stable under modification of the Pluscal code. To this end, we
+introduce a new module called `BakeryTyped.tla` with the following contents:
+
+```tla
+{{#include ../../../test/tla/bakery-pluscal/BakeryTyped.tla:1:26}}
+```
+
+Due to the semantics of `INSTANCE`, the constants and variables declared in
+`BakeryTyped.tla` substitute the constants and variables of
+`BakeryWoTlaps.tla`. By doing so we effectively introduce type annotations.
+Since we introduce a separate module, any changes in the Pluscal code do not
+affect our type annotations.
+
+Additionally, we add a constant initializer `ConstInit4`, which we will use
+later. See the manual section about the [ConstInit predicate][] for a detailed
+explanation.
+
+## Step 2: Annotate the operator `\prec`
+
+Let us run the type checker against `BakeryTyped.tla`:
+
+```sh
+$ apalache-mc typecheck BakeryTyped.tla
+...
+[BakeryWoTlaps.tla:66:17-66:20]: Cannot apply a to the argument 1 in a[1].
+...
+```
+
+The type checker complains about types of `a` and `b` in the operator `\prec`:
+
+```tla
+{{#include ../../../test/tla/bakery-pluscal/BakeryWoTlaps.tla:66:67}}
+```
+
+The issue is that the type checker is not able to decide, whether `a` and `b`
+are functions, sequences, or tuples. We help the type checker by adding type
+annotations to the operator `\preceq`.
```

Comment:
You mention earlier, that you don't want to add variable annotations, because of volatility. I'd spend a sentence to emphasize that you can/should annotate `\preceq`, because it's _not_ pluscal generated, and thus not volatile.